project/Build.scala

```diff
@@ -805,6 +805,74 @@ object Build {
+    withCommonSettings(Bootstrapped).
+    dependsOn(dottyCompiler(Bootstrapped) % "provided; compile->runtime; test->test").
+    dependsOn(`scala3-tasty-inspector` % "test->test").
+    settings(commonBootstrappedSettings).
+    settings(
+      javaOptions := (javaOptions in `scala3-compiler-bootstrapped`).value,
+      scalacOptions -= "-Xfatal-warnings",
+      ivyConfigurations += SourceDeps.hide,
+      transitiveClassifiers := Seq("sources"),
+      libraryDependencies +=
+        ("org.scala-lang" % "scala-library" % stdlibVersion(Bootstrapped) % "sourcedeps"),
+      sourceGenerators in Compile += Def.task {
+        val s = streams.value
+        val cacheDir = s.cacheDirectory
+        val trgDir = (sourceManaged in Compile).value / "scala-library-src"
+
+        val report = updateClassifiers.value
+        val scalaLibrarySourcesJar = report.select(
+            configuration = configurationFilter("sourcedeps"),
+            module = (_: ModuleID).name == "scala-library",
+            artifact = artifactFilter(`type` = "src")).headOption.getOrElse {
+          sys.error(s"Could not fetch scala-library sources")
+        }
+
+        FileFunction.cached(cacheDir / s"fetchScalaLibrarySrc",
+            FilesInfo.lastModified, FilesInfo.exists) { dependencies =>
+          s.log.info(s"Unpacking scala-library sources to $trgDir...")
+          if (trgDir.exists)
+            IO.delete(trgDir)
+          IO.createDirectory(trgDir)
+          IO.unzip(scalaLibrarySourcesJar, trgDir)
+
+          ((trgDir ** "*.scala") +++ (trgDir ** "*.java")).get.toSet
+        } (Set(scalaLibrarySourcesJar)).toSeq
+      }.taskValue,
+      sources in Compile ~= (_.filterNot(file =>
+        // sources from https://github.com/scala/scala/tree/2.13.x/src/library-aux
+        file.getPath.endsWith("scala-library-src/scala/Any.scala") ||
+        file.getPath.endsWith("scala-library-src/scala/AnyVal.scala") ||
+        file.getPath.endsWith("scala-library-src/scala/AnyRef.scala") ||
+        file.getPath.endsWith("scala-library-src/scala/Nothing.scala") ||
+        file.getPath.endsWith("scala-library-src/scala/Null.scala") ||
+        file.getPath.endsWith("scala-library-src/scala/Singleton.scala"))),
+      managedClasspath in Test ~= {
+        _.filterNot(file => file.data.getName == s"scala-library-${stdlibVersion(Bootstrapped)}.jar")
+      },
+    )
+
+  /** Test the tasty generated by `stdlib-bootstrapped
+   *
+   *  The tests are run with the bootstrapped compiler and the tasty inpector on the classpath.
+   *  The classpath has the default `scala-library` and not `stdlib-bootstrapped
+   *
+   *  The jar of `stdlib-bootstrappedovided for to the tests.
+   *   - inspector: test that we can load the contents of the jar using the tasty inspector
+   *   - from-tasty: test that we can recompile the contents of the jar using `dotc -from-tasty`
+   */
+  lazy val `stdlib-bootstrappedtests` = project.in(file("stdlib-stdlib-bootstrapped).
```

Comment:
I wouldn't call that project just stdlib-bootstrappedtests, because it's not clear how it's different from stdlib-bootstrapped/test, -tasty-test or -sanity-test would be better I think

Reply:
fixed